packages/fxa-auth-db-mysql/db-server/test/backend/db_tests.js

```diff
@@ -2597,6 +2597,84 @@ module.exports = function (config, DB) {
       })
     })
 
+    describe('account subscriptions', () => {
+      const subs = [...Array(10)].map(() => 'sub-' + Math.random())
+      let account
+
+      beforeEach(() => {
+        account = createAccount()
+        return db.createAccount(account.uid, account)
+      })
+
+      it('should create a new subscription for account', () => {
+        return db.createAccountSubscription(account.uid, subs[0], 'prod0', Date.now())
+          .then(result => assert.deepEqual(result, {}))
+      })
+
+      it('should fail to create for unknown account', () => {
+        return db.createAccountSubscription('12312312312', subs[1], 'prod2', Date.now())
+          .then(assert.fail, err => assert.equal(err.errno, 116, 'not found'))
+      })
+
+      it('should fail to create duplicate subscription id', () => {
+        const account2 = createAccount()
+        return db.createAccount(account2.uid, account2)
+          .then(() => db.createAccountSubscription(account.uid, subs[2], 'prod2', Date.now()))
+          .then(() => db.createAccountSubscription(account2.uid, subs[2], 'prod3', Date.now()))
+          .then(assert.fail, err => assert.equal(err.errno, 101, 'conflict'))
+      })
+
+      const pickSet = (list, name) => new Set(list.map(x => x[name]))
+
+      it('should support fetching all subscriptions', () =>
+        Promise.all([
+          db.createAccountSubscription(account.uid, subs[3], 'prod4', Date.now()),
+          db.createAccountSubscription(account.uid, subs[4], 'prod5', Date.now()),
+          db.createAccountSubscription(account.uid, subs[5], 'prod6', Date.now()),
+        ])
+        .then(() => db.fetchAccountSubscriptions(account.uid))
+        .then(result => {
+          assert.equal(result.length, 3)
```

Comment:
Can you use `assert.lengthOf` rather than `assert.equal` here?

Reply:
Oh weird, I hadn't seen `lengthOf` anywhere else in the tests so it didn't occur to me. 😓 Will change though.

Reply:
Ah, looks like I *can't* use `assert.lengthOf` here - [we use insist](https://github.com/mozilla/fxa/blob/c4c8571df870b104035518da1393c8b49a019160/packages/fxa-auth-db-mysql/db-server/test/backend/db_tests.js#L6) in these tests rather than Chai, and [insist](https://www.npmjs.com/package/insist) doesn't have `lengthOf`